### DISCUSSION TITLE
Can it be considered stable(functionality)?

### DISCUSSION THREAD

**Discussion by firefinchdev** [cashapp/molecule#62] [Q&A] [answered]
I know that latest release (0.1) does not signify stable release and that API may change considerably moving forward.
I want to know if Molecule is stable enough for production use (in terms of functionality, not the API).

I ask this because Molecule uses Compose engine(not Compose UI), which is already considered stable. As I read [here](https://code.cash.app/the-state-of-managing-state-with-compose), its already being used in Cash App.

Also, in terms of API, do you expect a lot to change inside the Composable functions? or is just the `Composable functions to Flow` API that may change?

**Comment by JakeWharton** [comment, marked as answer]
Yes, the quality of the library is production-ready and we are using it in production as you noted.

The version number significance is solely around the maturity and stability of the public API. We don't like launching libraries as 1.0 without first getting it in the hands of a bunch of people who are going to use it in ways we did not envision. For all we know, the public API may never change, or it will change completely.

We will do our best to use deprecation, maintain binary compatibility, and to offer refactoring suggestions if the API evolves as we try to in most our libraries.

**Reply by firefinchdev** [reply]
Okay thanks a lot :)
Just one thing..
When we say that "the API may/may not change in future", does that include things like LaunchEffect, etc (i.e. almost all the things used inside the composable function)?

I am assuming they wont change because they are part of compose engine (and thus common in both compose ui and Molecule)

**Reply by JakeWharton** [reply]
Correct. The only things that may change would be any of our API which is only three functions (well, more if you count the testing API). The Compose runtime is from Google and we inherit their stability guarantees.

**Reply by firefinchdev** [reply]
Makes sense.
Thanks for answering the queries.
I will try it, lets see if I get any ideas for improving.
